### PR TITLE
Plasma Core module development library

### PR DESCRIPTION
**Pull request by EatThePath**
Just a tiny little framework honest.

The Readme has lots of documentation in it. This is a first try of fennel/lua dual submission, so, I'm open to alternate ideas on how to structure it.

### CODE REVIEW

**Review by BMagnu — COMMENTED** [review]
The question here is, what does FSO do, when both the Fnl and the Lua version are present? Do they conflict?

**Comment by EatThePath** [comment]
If the Fennel package(sold separately) is not installed, .fnl files are invisible to`require` statements and thus only the lua will run. If the package is present then lua files will still be loaded by lua instead of .fnl files, so all should be good. After all, the Lua is equivelent to the code that the runtime would be executing anyay.

...except the modular configs. Drat. I'm calling the fennel compiler to load those, so it's required rather than optional. Hm.

**Comment by EatThePath** [comment]
I've got an idea on how to handle configs that'll slightly get around this, so drafting for now

**Comment by EatThePath** [comment]
The config loading is now flexible, and a lua-only solution is provided for people who want to go that route.

File: plasma_core/data/scripts/repl.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  if (_2_ == "1") then
+    return dj:play_new("DawnWave1Loop", "HardMeasure")
+  elseif (_2_ == "2") then
+    return dj:play_new("DawnWave2Main", "ExitLoop")
+  elseif (_2_ == "8") then
+    return core:reload()
+  elseif (_2_ == "9") then
+    return core.recursive_table_print("state", dj)
+  elseif (_2_ == "0") then
+    return engine.restartLog()
```
Comment (BMagnu) [inline plasma_core/data/scripts/repl.lua]: Looking at the above Fennel version, this should likely be regenerated to respect the commented out lines regarding the dj-script (which shouldn't be loaded given its not present in these files)

File: plasma_core/data/scripts/plasma_core.fnl [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+  (let [label (if (= nil ?label) "" ?label)
+        t (type output)
+        has_label (< 0 (length label))
+        core _G.plasma_core]
+  (match (type output)
+    :table (core.recursive_table_print label output)
+    :userdata (ba.print (.. "*: " label (if has_label " " "")  "is userdata\n"))
+    :string (ba.print (.. "*: " label (if has_label " " "")  output "\n"))
+    :Nil (ba.print (.. "*: " label (if has_label " " "")  "is nil" "\n"))
+    :_ (ba.print (.. "*: " label (if has_label" " "") "type " t ":" (_G.totring output) "\n")))))
```
Comment (BMagnu) [inline plasma_core/data/scripts/plasma_core.fnl]: THis should be tostring, no?

**Review by BMagnu — CHANGES_REQUESTED** [review]
Small comments

File: plasma_core/data/scripts/plasma_core.fnl (hunk 2 above)
Comment (BMagnu) [reply plasma_core/data/scripts/plasma_core.fnl]: plus regenerating the corresponding lua file, of course

**Review by BMagnu — APPROVED** [review]

**Pull request merged**